src/Logic.hs

```diff
@@ -360,27 +373,26 @@ handlePullRequestEdited prId newTitle newBaseBranch state =
 -- indicate the `Merge` approval type.
 parseMergeCommand :: TriggerConfiguration -> Text -> Maybe ApprovedFor
 parseMergeCommand config message =
-  let
-    messageCaseFold = Text.toCaseFold $ Text.strip message
-    prefixCaseFold = Text.toCaseFold $ Config.commentPrefix config
-  in
-    -- Check if the prefix followed by ` merge and {deploy,tag}` occurs within the message.
-    -- We opt to include the space here, instead of making it part of the
-    -- prefix, because having the trailing space in config is something that is
-    -- easy to get wrong.
-    -- Note that because "merge" is an infix of "merge and xxx" we need to
-    -- check for the "merge and xxx" commands first: if this order were
-    -- reversed all "merge and xxx" commands would be detected as a Merge
-    -- command.
-    if (prefixCaseFold <> " merge and deploy") `Text.isInfixOf` messageCaseFold
-    then Just MergeAndDeploy
-    else
-      if (prefixCaseFold <> " merge and tag") `Text.isInfixOf` messageCaseFold
-      then Just MergeAndTag
-      else
-        if (prefixCaseFold <> " merge") `Text.isInfixOf` messageCaseFold
-        then Just Merge
-        else Nothing
+  let messageCaseFold = Text.toCaseFold $ Text.strip message
+      prefixCaseFold = Text.toCaseFold $ Config.commentPrefix config
+      infixMatch msg = (prefixCaseFold <> msg) `Text.isInfixOf` messageCaseFold
+      -- Check if the prefix followed by ` merge [and {deploy,tag}] [on friday]` occurs within the message.
+      -- We opt to include the space here, instead of making it part of the
+      -- prefix, because having the trailing space in config is something that is
+      -- easy to get wrong.
+      -- Note that because "merge" is an infix of "merge and xxx" we need to
+      -- check for the "merge and xxx" commands first: if this order were
+      -- reversed all "merge and xxx" commands would be detected as a Merge
+      -- command.
+      cases =
+        [ (" merge and deploy on friday", MergeAndDeployOnFriday),
```

Comment:
A minor thing: I wonder if "on" is a good choice of words here, as it could be mistaken for a scheduled merge (which we don't support). Maybe "even on"? Not really sure, would otherwise also be fine.